app/services/puppet_fact_parser.rb

```diff
@@ -185,7 +185,13 @@ def use_legacy_facts?
   end
 
   def os_name
-    facts[:operatingsystem].presence || raise(::Foreman::Exception.new("invalid facts, missing operating system value"))
+    os_name = facts[:operatingsystem].presence || raise(::Foreman::Exception.new("invalid facts, missing operating system value"))
+
+    if os_name == 'RedHat' && facts[:lsbdistid] == 'RedHatEnterpriseWorkstation'
```

Comment:
Can you check `facts.dig(:os).dig(:distro).dig(:id)` first? I believe the `lsbdistid` is a legacy fact that will be dropped eventually.

Reply:
Indeed this is now just an alias in facter-ng, might get dropped: https://github.com/puppetlabs/facter-ng/blob/30b2739963eb1251b4f24ed210155d4f02f228ee/lib/facts/debian/os/distro/id.rb

Reply:
It is indeed a legacy fact: https://puppet.com/docs/facter/3.11/core_facts.html#lsbdistid

Reply:
I know it's legacy one, I did that on purpose to avoid mixing legacy in this file with nonlegacy, we already use :lsbdistcodename, :lsbdistdescription, :lsbdistrelease already

however I can change that if you think it's better, just please comment "yes" once again :-)

Reply:
Yeah that should be probably fixed too. Actually, I don't mind. Dropping facter 2 is probably a different story we need to work on.